components/brave_rewards/resources/extension/brave_rewards/components/panel.tsx

```diff
@@ -335,8 +336,19 @@ export class Panel extends React.Component<Props, State> {
     }
   }
 
+  refreshPublisher = () => {
+    const publisher: RewardsExtension.Publisher | undefined = this.getPublisher()
+    const publisherKey = publisher && publisher.publisher_key
+    chrome.tabs.query({
```

Comment:
you don't need to query tab, you can get window id in `this.state.windowId`

Reply:
It's in `this.props.windowId`, but thanks. Good to know.

Reply:
sorry, it's in `state` in `app.tsx` and in `props` in `panel.tsx`

Reply:
Done.